src/components/documents/CmsText/WithCmsText.tsx

```diff
@@ -0,0 +1,19 @@
+import { CmsTextContext, useFirstCmsTextDocumentIfExists } from '@tdev-components/documents/CmsText/shared';
+import { observer } from 'mobx-react-lite';
+
+interface Props {
+    id: string;
+    children?: React.ReactNode;
+}
+
+const WithCmsText = observer(({ id, children }: Props) => {
+    const doc = useFirstCmsTextDocumentIfExists(id);
+
+    return doc ? (
+        <CmsTextContext.Provider value={{ cmsText: doc.text }}>{children}</CmsTextContext.Provider>
```

Comment:
Idea: When we pass down the `cmsTextId` we can have a single logic inside the `CmsText`-Component. The overhead should be negligible (mobx computed's). (I have no strong opinion here, but maybe the code gets cleaner?)